### PR TITLE
activate derive serde feature for halo2_proofs

### PR DESCRIPTION
**Pull request by zemse**
Activates derive_serde feature for halo2_proofs.

### CODE REVIEW

File: halo2-base/src/utils/halo2.rs [hunk 1]
```diff
@@ -175,7 +176,7 @@ mod keygen {
             let pk = plonk::keygen_pk2(kzg_params, &circuit, false).unwrap();
             #[cfg(not(feature = "halo2-axiom"))]
             let pk = {
-                let vk = plonk::keygen_vk_custom(kzg_params, &circuit, false).unwrap();
+                let vk = plonk::keygen_vk(kzg_params, &circuit).unwrap();
```
Comment (zemse) [inline halo2-base/src/utils/halo2.rs:179]: Highlighting this change, since I couldnt find `keygen_vk_custom`, set to `keygen_vk` for now.

**Comment by jonathanpwang** [comment]
Updated pse `halo2_proofs` to `v0.3.0`. This should be fixed on `main` now. Please close if you can confirm.

**Comment by zemse** [comment]
Previous changes were not needed. I've updated the PR

**Review by jonathanpwang — APPROVED** [review]
LGTM

**Pull request merged**